python_modules/libraries/dagster-airbyte/dagster_airbyte/asset_defs.py

```diff
@@ -237,19 +252,29 @@ def _get_normalization_tables_for_schema(
 
     out = []
     # Object types are broken into a new table, as long as they have children
-    if (
-        schema["type"] == "object"
-        or "object" in schema["type"]
-        and len(schema.get("properties", {})) > 0
-    ):
-        out.append(prefix + key)
-        for k, v in schema["properties"].items():
-            out += _get_normalization_tables_for_schema(k, v, f"{prefix}{key}_")
-    # Array types are also broken into a new table
-    elif schema["type"] == "array" or "array" in schema["type"]:
-        out.append(prefix + key)
-        for k, v in schema["items"]["properties"].items():
-            out += _get_normalization_tables_for_schema(k, v, f"{prefix}{key}_")
+
+    sub_schemas = _get_sub_schemas(schema)
+
+    for sub_schema in sub_schemas:
+        schema_type = _get_schema_types(sub_schema)
```

Comment:
this logic might be slightly simpler if _get_schema_types returns Optional[List[str]] (rather than either str, List[str], or None).